src/vmm/src/signal_handler.rs

```diff
@@ -166,6 +200,16 @@ impl SignalManager {
         .into_result()
         .map_err(Error::SyscallError)?;
 
+        // SIGSYS is handled separately because it doesn't have proper
+        // signalfd support in older kernels.
+        //
+        // Call to register_signal_handler is considered unsafe because it will register a signal
+        // handler which will be called in the current thread and will interrupt whatever work
+        // is done on the current thread, so we have to keep in mind that the registered
+        // signal handler must only do async-signal-safe operations.
+        register_signal_handler(SIGSYS, sigsys_handler)
```

Comment:
The `register_signal_handler` call is out of scope for this function, which should only return the mask used for blocking other signals.

I suggest we call this `register_signal_handler` function explicitly in the `main.rs`, before spawning any threads, to make sure that they inherit the handler